pkg/cmd/cli/cmd/set/probe.go

```diff
@@ -0,0 +1,409 @@
+indicate when your container is ready to accept incoming traffic or begin handling work.
+Setting both liveness and readiness probes for each container is highly recommended.
+
+The three probe types are:
+
+1. Open a TCP socket on the pod IP
+2. Perform an HTTP GET against a URL on a container that must return 200 OK
+3. Run a command in the container that must return exit code 0
+
+Containers that take a variable amount of time to start should set generous
+initial-delay-seconds values, otherwise as your application evolves you may suddenly begin
+to fail.`
+
+	probeExample = `  # Clear both readiness and liveness probes off all containers
+  $ %[1]s probe dc/registry --unset --readiness --liveness
+
+  # Set an exec action as a liveness probe to run 'echo ok'
+  $ %[1]s probe dc/registry --liveness -- echo ok
+
+  # Set a readiness probe to try to open a TCP socket on 3306
+  $ %[1]s probe rc/mysql --readiness --open-tcp=3306
+
+  # Set an HTTP readiness probe for port 8080 and path /healthz over HTTP on the pod IP
+  $ %[1]s probe dc/webapp --readiness --get-url=http://:8080/healthz
+
+  # Set an HTTP readiness probe over HTTPS on 127.0.0.1 for a hostNetwork pod
+  $ %[1]s probe dc/router --readiness --get-url=https://127.0.0.1:1936/stats
+
+  # Set only the initial-delay-seconds field on all deployments
+  $ %[1]s probe dc --all --readiness --initial-delay-seconds=30`
+)
+
+type ProbeOptions struct {
+	Out io.Writer
+	Err io.Writer
+
+	Filenames         []string
+	ContainerSelector string
+	Selector          string
+	All               bool
+
+	Builder *resource.Builder
+	Infos   []*resource.Info
+
+	Encoder runtime.Encoder
+
+	ShortOutput bool
+	Mapper      meta.RESTMapper
+
+	PrintObject            func(runtime.Object) error
+	UpdatePodSpecForObject func(runtime.Object, func(spec *kapi.PodSpec) error) (bool, error)
+
+	Readiness bool
+	Liveness  bool
+	Unset     bool
+
+	OpenTCPSocket string
+	HTTPGet       string
+	Command       []string
+
+	FlagSet       func(string) bool
+	HTTPGetAction *kapi.HTTPGetAction
+
+	// Length of time before health checking is activated.  In seconds.
+	InitialDelaySeconds *int
+	// Length of time before health checking times out.  In seconds.
+	TimeoutSeconds *int
+	// How often (in seconds) to perform the probe.
+	PeriodSeconds *int
+	// Minimum consecutive successes for the probe to be considered successful after having failed.
+	// Must be 1 for liveness.
+	SuccessThreshold *int
+	// Minimum consecutive failures for the probe to be considered failed after having succeeded.
+	FailureThreshold *int
+}
+
+// NewCmdProbe implements the set probe command
+func NewCmdProbe(fullName string, f *clientcmd.Factory, out, errOut io.Writer) *cobra.Command {
+	options := &ProbeOptions{
+		Out: out,
+		Err: errOut,
+
+		ContainerSelector: "*",
+	}
+	cmd := &cobra.Command{
+		Use:     "probe RESOURCE/NAME [--readiness|--liveness] ...",
```

Comment:
Probably. Resource arg can be just `RESOURCE/NAME`, we do that already in other commands even when they accept `--all`. If the rest fits it would be good, makes it pretty clear what options you have.